design/witx-type-representation.md

```diff
@@ -0,0 +1,86 @@
+# Witx Type Representation
+In Witx, each type corresponds to a lower-level type. For example, the `size` type resolves to a `u32`, the `errno` type resolves to an `Enum(u16)`, and more. Detailed below is each type, and what they resolve to in memory.
```

Comment:
"what they resolve to in memory" seems a little odd.  How about "what the correspond to in the WebAssembly type type system"?

Reply:
"WebAssembly type type" system seems like a typo, so I changed that to "WebAssembly type system".